### ISSUE TITLE
Annotation of multiple properties broken since SMW 2.3

### ISSUE THREAD

**Issue by krabina** [SemanticMediaWiki/SemanticMediaWiki#1252]
The annotation of multiple properties described in [1] ("assigning one value to multipel properties") is broken since SMW 2.3. It can be seen here: [2].

The annotation ``[[Testproperty1::Testproperty2::200]]`` does not work: Testproperty1 is set to "Testproperty2::200" instead of Testproperty1 and Testproperty2 both set to "200".

[1] https://semantic-mediawiki.org/wiki/Help:In-text_annotation
[2] http://sandbox.semantic-mediawiki.org/wiki/Testproperty

**Comment by mwjames** [comment]
> he annotation [[Testproperty1::Testproperty2::200]] 

@kghbln Do we really support this?? This is __not__ tested at all because we explicitly fixed `::` annotation use (#1066, #1075) so we know that the first part is a property and the rest belongs to a value `[[DOI::10.1002/123::abc]]` or `[[Foo:::123]]`.

We have a test case `[[Has page:::Page/::Foo:Bar]] [[Has text:::Text/::Foo:Bar]] [[Has url::http://example.org/::Foo:Bar]]` where we specifically test that values contain ` "Page/::Foo:Bar", ":Text/::Foo:Bar", "http://example.org/::Foo:Bar"`

If this is really a supported use case then we need to add a config setting for `[[property::property::property::value]]` to explicitly enable this as we have no way of telling what is or isn't a property. I also would make this setting an exception rather then a default because `[[ ... :: ... ]]]` is to declare one triple. (subject -> property -> value).

**Comment by krabina** [comment]
It worked in the previous version and it is documented in the user manual...

**Comment by kghbln** [comment]
This possibility to assign one value to multiple properties was there from the/my start so SMW 1.4 minimum so this is indeed a regression. Personally I have never used this so it did not come to my mind when you added the new features but it may very well be out there and we cannot blame them to have used it. I'd opt for a configuration setting which defaults to "false". "true" would allow to migrate or willingly use it as an exception.

**Comment by krabina** [comment]
My current use case is to use it in an arraymap:
{{#arraymap:{{{Benutzer|}}}|,|x|[[Benutzer::Assigned to::Benutzer:x]]}}
This is so I can set a property of my choice alongside the property assigned to that is defined by Semantic Tasks. But I just tested it, and 
{{#arraymap:{{{Benutzer|}}}|,|x|[[Benutzer::Benutzer:x]][[Assigned to::Benutzer:x]]}}
works fine as well...

**Comment by mwjames** [comment]
> Personally I have never used this so it did not come to my mind when you added the new features 

Same here and that is why we have a test for the feature but not for "many to one" interpretation.

```
##
# The strict mode is to remove [[ ... :: ... ]] ambiguity during
# the string parsing process.
# 
# The default interpretation (strict) is [[property::value:partOfTheValue::alsoPartOfTheValue]] but
# if the strict mode is disabled then multiple properties can be assigned to one value
# [[property1::property2::value]] which may cause value strings to interpret unanticipated
# in case of additional colons. 
#
# @since 2.3
##
$GLOBALS['smwgInTextAnnotationParserStrictMode'] = true;
```

Does this make sense? The effective change would be a one line but since I have to write tests that doesn't happen in the blink of an eye.

PS: I really hope we would have tests before fixing or adding features that would make life so much easier knowing that something is broken, now we have to work backwards.

**Comment by JeroenDeDauw** [comment]
If it makes you feel any better, I had no idea this feature existed either :)

**Comment by mwjames** [comment]
@krabina Added your test case in `p-0408 [#1252] intext to use non-strict mode.json` with `$GLOBALS['smwgEnabledInTextAnnotationParserStrictMode']` set to `false` to support `[[Testproperty1::Testproperty2::200]]`.

[0] will be available with the next 2.3.1 release.

[0] https://semantic-mediawiki.org/wiki/Help:$smwgEnabledInTextAnnotationParserStrictMode
